Projects/b_u585i_iot02a_ntz/Src/ota_pal/ota_pal_stm32u5_ntz.c

```diff
@@ -47,150 +47,306 @@
 
 #include "PkiObject.h"
 
-const char OTA_JsonFileSignatureKey[ OTA_FILE_SIG_KEY_STR_MAX_LENGTH ] = "sig-sha256-ecdsa";
-
-#define FLASH_START_CURRENT_BANK    ( ( uint32_t ) FLASH_BASE )
-
-#define FLASH_START_PASSIVE_BANK    ( ( uint32_t ) ( FLASH_BASE + FLASH_BANK_SIZE ) )
+#define FLASH_START_INACTIVE_BANK    ( ( uint32_t ) ( FLASH_BASE + FLASH_BANK_SIZE ) )
 
 #define NUM_QUAD_WORDS( length )         ( length >> 4UL )
 
 #define NUM_REMAINING_BYTES( length )    ( length & 0x0F )
 
-#define ALTERNATE_BANK( bank )           ( ( bank == FLASH_BANK_1 ) ? FLASH_BANK_2 : FLASH_BANK_1 )
-
 #define IMAGE_CONTEXT_FILE_NAME    "/ota/image_state"
 
-typedef enum OtaPalImageStateInternal
+#define OTA_IMAGE_MIN_SIZE         ( 16 )
+
+
+typedef enum
```

Comment:
It will be good to document all the states and the possible transitions to different states.